### ISSUE TITLE
Migration to .NET 5.0

### ISSUE THREAD

**Issue by Samuel12321** [ShankarBUS/ModernFlyouts#28]
We are currently targeting .NET Framework 4.6.2 and .NET Core 3.1

It would be simpler and faster to just use one framework/runtime, i.e. .NET 5. It seems to be a better solution.

What do you think? @ShankarBUS

**Comment by ShankarBUS** [comment]
We are not moving to UWP, we are already using some UWP (i.e. WinRT) APIs such as **GlobalSessionMediaTransportControls**.

We are going to make the app packaged/app-container compatible (not only the store version but also the Github version), so that both versions will behave same (we can remove the current startup method and app configs which will eliminate those two recent bugs). We will have some advantages like toast notifications, start menu jump lists (we can replace the tray icon with this), startup task, app configs which the unpackaged apps won't.

I was waiting for .NET 5 to have public availability and it seems now is the time 😀. I don't know how we can enable framework packages so that store will install .NET 5 runtime for us and update it automatically. Will dig into that.

**Comment by ShankarBUS** [comment]
And another issue with .NET Core 3.0/.NET 5.0 is the app package size 😔. We have to bundle the .NET runtime with the app or make the user download the latest runtime. (Which won't be a problem with .NET Framework since it is readily available in windows 10) We have to find a way on how some UWP apps use framework packaging to tackle the .NET Native installation. We can replicate that with .NET 5

**Comment by ShankarBUS** [comment]
I don't know if .NET 5 is supported by MSIX.

But if so, [.NET 5 has an incredible way to trim dependency size](https://devblogs.microsoft.com/dotnet/app-trimming-in-net-5/)

**Comment by Samuel12321** [comment]
.NET 5 is support has been added to MSIX on beta Visual Studio

**Comment by ShankarBUS** [comment]
Wow! that's awesome. We now just have to wait for our dependencies to target .NET 5

**Comment by Samuel12321** [comment]
just created a very rough test .net5 build that seems to work in my fork in the .net5 testing branch

**Comment by ShankarBUS** [comment]
Perfect! What about the publish issues?

**Comment by ShankarBUS** [comment]
Could we switch to .NET 5 immediately. Because we may have to be concrete on what runtime we choose before I start working on v0.7

Are there any issues with the .NET 5 build and packaging?

**Comment by Samuel12321** [comment]
Performing well, only issue is a reoccurrence of the can only build for x86 bug

**Comment by Samuel12321** [comment]
> Could we switch to .NET 5 immediately. 
yes if we fix the x86 only bug

We could do a 0.6.5 release with the update being switching to .net5

**Comment by ShankarBUS** [comment]
So the first thing to do before starting v0.7 work will be to switch to .NET 5, right?

**Comment by Samuel12321** [comment]
yes i think so we can have a minor 0.6.5 release switching to .net 5

**Comment by Samuel12321** [comment]
Just released 0.6.5.
Still a few more things to do to finalise .NET5 integration:
 -  convert to .dll to fix x86 issue
-  fix issue that is preventing implementation of `PublishTrimmed`

**Comment by Samuel12321** [comment]
.dll injection canned as can not be done with .NET
